code/datums/weather/weather.dm

```diff
@@ -105,28 +146,68 @@
 /datum/weather/proc/telegraph()
 	if(stage == STARTUP_STAGE)
 		return
-	SEND_GLOBAL_SIGNAL(COMSIG_WEATHER_TELEGRAPH(type), src)
 	stage = STARTUP_STAGE
+	setup_weather_areas(impacted_areas)
+
+	if(weather_flags & (WEATHER_TURFS|WEATHER_THUNDER))
+		setup_weather_turfs()
+
+	SEND_GLOBAL_SIGNAL(COMSIG_WEATHER_TELEGRAPH(type), src)
+
+	weather_duration = rand(weather_duration_lower, weather_duration_upper)
+	SSweather.processing |= src
+	update_areas()
+	if(telegraph_duration)
+		send_alert(telegraph_message, telegraph_sound, telegraph_sound_vol)
+	addtimer(CALLBACK(src, PROC_REF(start)), telegraph_duration)
+
+/datum/weather/proc/setup_weather_areas(list/selected_areas)
+	if(length(selected_areas))
+		return // impacted areas already been setup
+
```

Comment:
I'm sorry this pattern stinks please just return a list of selected areas pass by ref is HELL

Reply:
I'll see if I can rework it.

Reply:
This was refactored and removed via https://github.com/tgstation/tgstation/commit/05e0edb485b6a57d294186401c4435308e1782d4